<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2020-04-26" />

<title>Download data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bgee</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Contents</a>
</li>
<li>
  <a href="general.html">Software &amp; Data</a>
</li>
<li>
  <a href="extractinfo.html">Extract Information</a>
</li>
<li>
  <a href="downloaddata.html">Download data</a>
</li>
<li>
  <a href="processdata.html">Processing data</a>
</li>
<li>
  <a href="topanat.html">TopAnat</a>
</li>
<li>
  <a href="sparql.html">Sparql endpoint</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/SFonsecaCosta/Bgee">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Download data</h1>
<h4 class="author">Sara Fonseca Costa</h4>
<address class="author_afil">
Department of ecology and evolution, University of Lausanne &amp; Swiss Institute of Bioinformatics, Switzerland<br><h4 class="date">April 26, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-04-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Bgee/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200417code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200417)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200417code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200417)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomSFonsecaCostaBgeetreec287d019b34ad31c557449f32e0d1c221e15df37targetblankc287d01a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/SFonsecaCosta/Bgee/tree/c287d019b34ad31c557449f32e0d1c221e15df37" target="_blank">c287d01</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomSFonsecaCostaBgeetreec287d019b34ad31c557449f32e0d1c221e15df37targetblankc287d01a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/SFonsecaCosta/Bgee/tree/c287d019b34ad31c557449f32e0d1c221e15df37" target="_blank">c287d01</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:  analysis/.here

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/downloaddata.Rmd</code>) and HTML (<code>docs/downloaddata.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SFonsecaCosta/Bgee/blob/c287d019b34ad31c557449f32e0d1c221e15df37/analysis/downloaddata.Rmd" target="_blank">c287d01</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-26
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SFonsecaCosta/Bgee/ae29961e3b7c4f01a40198e3dcf18f03e8ebdde8/docs/downloaddata.html" target="_blank">ae29961</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SFonsecaCosta/Bgee/blob/9907294c07372d3e907a7d584449805c765aaeab/analysis/downloaddata.Rmd" target="_blank">9907294</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-22
</td>
<td>
clean text
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SFonsecaCosta/Bgee/8d9e88f2c220aab30cd9e8a10f480180c3de1d51/docs/downloaddata.html" target="_blank">8d9e88f</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SFonsecaCosta/Bgee/blob/f18beb56d5f4ad2df3b91c03b4873c6ecbce00a7/analysis/downloaddata.Rmd" target="_blank">f18beb5</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-20
</td>
<td>
Minor things
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SFonsecaCosta/Bgee/3118c12fb298dfed29663b63d4e757672d38fbc9/docs/downloaddata.html" target="_blank">3118c12</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SFonsecaCosta/Bgee/blob/4b9ee875926f5fcdce75eee565b4f2b458791149/analysis/downloaddata.Rmd" target="_blank">4b9ee87</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-20
</td>
<td>
cache=FALSE
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SFonsecaCosta/Bgee/8d821e2b4d95ed6dcc78135555e80427a7c48dd9/docs/downloaddata.html" target="_blank">8d821e2</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SFonsecaCosta/Bgee/blob/9073f839d6190c28fa785eecc15e12e945fdc57b/analysis/downloaddata.Rmd" target="_blank">9073f83</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-20
</td>
<td>
add analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="all-experiments" class="section level2">
<h2>All experiments</h2>
<p>If you do not want to use filters, as showed in the previous section (Extract Information) where you target a particular experiment to download, you can always download all the data of reference species using the <code>getData()</code> function. Note that in the getData() you should specify which data type you would like to retrieve from Bgee by specifying that in the argument <code>dataType</code>.</p>
<pre class="r"><code>DrosMelRNASeq &lt;- Bgee$new(species = &quot;Drosophila_melanogaster&quot;, dataType = &quot;rna_seq&quot;)</code></pre>
<pre><code>
Querying Bgee to get release information...

Building URL to query species in Bgee release 14_1...

Submitting URL to Bgee webservice... (https://r.bgee.org/bgee14_1/?page=r_package&amp;action=get_all_species&amp;display_type=tsv&amp;source=BgeeDB_R_package&amp;source_version=2.12.1)

Query to Bgee webservice successful!

API key built: 342617eeb6e728567eeaf07855efc4fa274e6ad21dde329ecce7bd3668c4efa9f23c364808e45b7b772594a592dfe6855ff75027e7983a848ff43a25416ebe6f</code></pre>
<pre class="r"><code>dataBgee_DM &lt;- getData(DrosMelRNASeq)</code></pre>
<pre><code>The experiment is not defined. Hence taking all rna_seq experiments available for Drosophila_melanogaster.</code></pre>
<pre><code>Downloading expression data...</code></pre>
<pre><code>Saved expression data file in/Users/sfonseca1/Bgee/Drosophila_melanogaster_Bgee_14_1 folder. Now untar file...</code></pre>
<pre><code>Finished uncompress tar files</code></pre>
<pre><code>Saving all data in .rds file...</code></pre>
<pre class="r"><code>## Number of experiments that exist in BgeeDB
length(dataBgee_DM)</code></pre>
<pre><code>[1] 11</code></pre>
</div>
<div id="subset-of-the-data" class="section level2">
<h2>Subset of the data</h2>
<p>In case you are interested in downloading a particular experiment (as showed before), anatomical entity or developmental stage, you can just download this data by specifying that in the respective arguments: <code>experimentId</code> , <code>anatEntityId</code> and <code>stageId</code>.</p>
<div id="per-experiment" class="section level3">
<h3>Per experiment</h3>
<p>Taking your experimentId result from the filtering in previously section, download just this respective experiment by specifying that in the argument <code>experimentId</code> in the <code>getData()</code> function.</p>
<pre class="r"><code>DrosMelRNASeq_SRP002072 &lt;- getData(DrosMelRNASeq, experimentId = &quot;SRP002072&quot;)</code></pre>
<pre><code>Downloading expression data for the experiment SRP002072...</code></pre>
<pre><code>Saved expression data file in /Users/sfonseca1/Bgee/Drosophila_melanogaster_Bgee_14_1 folder. Now untar file...</code></pre>
<pre><code>Finished uncompress tar files</code></pre>
<pre><code>Saving all data in .rds file...</code></pre>
<pre class="r"><code>head(DrosMelRNASeq_SRP002072)</code></pre>
<pre><code>  Experiment.ID Library.ID Library.type     Gene.ID Anatomical.entity.ID
1     SRP002072  SRX019645       paired FBgn0000003       UBERON:0000922
2     SRP002072  SRX019645       paired FBgn0000008       UBERON:0000922
3     SRP002072  SRX019645       paired FBgn0000014       UBERON:0000922
4     SRP002072  SRX019645       paired FBgn0000015       UBERON:0000922
5     SRP002072  SRX019645       paired FBgn0000017       UBERON:0000922
6     SRP002072  SRX019645       paired FBgn0000018       UBERON:0000922
  Anatomical.entity.name      Stage.ID                     Stage.name  Sex
1                 embryo FBdv:00005306 embryonic stage 4 (Drosophila) &lt;NA&gt;
2                 embryo FBdv:00005306 embryonic stage 4 (Drosophila) &lt;NA&gt;
3                 embryo FBdv:00005306 embryonic stage 4 (Drosophila) &lt;NA&gt;
4                 embryo FBdv:00005306 embryonic stage 4 (Drosophila) &lt;NA&gt;
5                 embryo FBdv:00005306 embryonic stage 4 (Drosophila) &lt;NA&gt;
6                 embryo FBdv:00005306 embryonic stage 4 (Drosophila) &lt;NA&gt;
  Strain Read.count         TPM         FPKM Detection.flag Detection.quality
1     NA  5071.0000 7931.226169 10518.520218        present      high quality
2     NA   172.0002    5.951876     7.893474        present      high quality
3     NA    16.0000    0.604627     0.801867        present      high quality
4     NA    19.0000    0.793388     1.052205        present      high quality
5     NA   841.1847   22.824416    30.270109        present      high quality
6     NA    94.0000    9.765199    12.950765        present      high quality
   State.in.Bgee
1 Part of a call
2 Part of a call
3 Part of a call
4 Part of a call
5 Part of a call
6 Part of a call</code></pre>
<p>Note that in the next sub-sections the arguments <code>anatEntityId</code> and <code>stageId</code> in the getData() function are implemented just in BgeeDB package in the Bioconductor release 3.11.</p>
</div>
<div id="specify-an-anatomical-entity" class="section level3">
<h3>Specify an anatomical entity</h3>
<p>You are also able to retrieve data from the database by specifying the anatomic entity of interest for this particular experiment. Here we will target adult organism as an example.</p>
<pre class="r"><code>DrosMelRNASeq_annEnt &lt;- getData(DrosMelRNASeq, experimentId = &quot;SRP002072&quot;, anatEntityId = &quot;UBERON:0007023&quot;)
head(DrosMelRNASeq_annEnt)</code></pre>
<p>Note you can specify more than one anatomical entity in the <code>anatEntityId</code> argument.</p>
<pre class="r"><code>DrosMelRNASeq_annEnt_plus &lt;- getData(DrosMelRNASeq, experimentId = &quot;SRP002072&quot;, anatEntityId = c(&quot;UBERON:0007023&quot;,&quot;UBERON:0000922&quot;,&quot;UBERON:0002548&quot;))
head(DrosMelRNASeq_annEnt_plus)</code></pre>
</div>
<div id="specify-a-developmental-stage" class="section level3">
<h3>Specify a developmental stage</h3>
<p>As demonstrated before, for experiment and anatomical entities, you also can download data by specifying your target developmental stage of interest in the <code>stageId</code> argument.</p>
<pre class="r"><code>DrosMelRNASeq_SRP002072_stg &lt;- getData(DrosMelRNASeq, stageId = c(&quot;UBERON:0000068&quot;,&quot;FBdv:00005341&quot;))
head(DrosMelRNASeq_SRP002072_stg)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.0 (2019-04-26)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] BgeeDB_2.12.1        tidyr_1.0.2          topGO_2.38.1        
 [4] SparseM_1.78         GO.db_3.10.0         AnnotationDbi_1.48.0
 [7] IRanges_2.20.2       S4Vectors_0.24.4     Biobase_2.46.0      
[10] graph_1.64.0         BiocGenerics_0.32.0  workflowr_1.6.1     

loaded via a namespace (and not attached):
 [1] tidyselect_1.0.0   xfun_0.13          purrr_0.3.3        lattice_0.20-41   
 [5] vctrs_0.2.4        htmltools_0.4.0    yaml_2.2.1         blob_1.2.1        
 [9] rlang_0.4.5        later_1.0.0        pillar_1.4.3       glue_1.4.0        
[13] DBI_1.1.0          bit64_0.9-7        matrixStats_0.56.0 lifecycle_0.2.0   
[17] stringr_1.4.0      memoise_1.1.0      evaluate_0.14      knitr_1.28        
[21] httpuv_1.5.2       fansi_0.4.1        Rcpp_1.0.4.6       promises_1.1.0    
[25] backports_1.1.6    fs_1.4.1           bit_1.1-15.2       digest_0.6.25     
[29] stringi_1.4.6      dplyr_0.8.5        rprojroot_1.3-2    grid_3.6.0        
[33] cli_2.0.2          tools_3.6.0        bitops_1.0-6       magrittr_1.5      
[37] RCurl_1.98-1.1     RSQLite_2.2.0      tibble_3.0.0       crayon_1.3.4      
[41] whisker_0.4        pkgconfig_2.0.3    ellipsis_0.3.0     data.table_1.12.8 
[45] assertthat_0.2.1   rmarkdown_2.1      R6_2.4.1           git2r_0.26.1      
[49] compiler_3.6.0    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
