<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2020-04-22" />

<title>Extract information</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bgee</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Contents</a>
</li>
<li>
  <a href="general.html">Software &amp; Data</a>
</li>
<li>
  <a href="extractinfo.html">Extract Information</a>
</li>
<li>
  <a href="downloaddata.html">Download data</a>
</li>
<li>
  <a href="processdata.html">Processing data</a>
</li>
<li>
  <a href="topanat.html">TopAnat</a>
</li>
<li>
  <a href="sparql.html">Sparql endpoint</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/SFonsecaCosta/Bgee">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Extract information</h1>
<h4 class="author">Sara Fonseca Costa</h4>
<address class="author_afil">
Department of ecology and evolution, University of Lausanne, Switzerland<br><h4 class="date">April 22, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-04-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Bgee/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200417code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200417)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200417code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200417)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomSFonsecaCostaBgeetree9907294c07372d3e907a7d584449805c765aaeabtargetblank9907294a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/SFonsecaCosta/Bgee/tree/9907294c07372d3e907a7d584449805c765aaeab" target="_blank">9907294</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomSFonsecaCostaBgeetree9907294c07372d3e907a7d584449805c765aaeabtargetblank9907294a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/SFonsecaCosta/Bgee/tree/9907294c07372d3e907a7d584449805c765aaeab" target="_blank">9907294</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:  Bos_taurus_Bgee_14_1/
    Untracked:  Drosophila_melanogaster_Bgee_14_1/
    Untracked:  release.tsv
    Untracked:  species_Bgee_14_1.tsv

Unstaged changes:
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/license.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/extractinfo.Rmd</code>) and HTML (<code>docs/extractinfo.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SFonsecaCosta/Bgee/blob/9907294c07372d3e907a7d584449805c765aaeab/analysis/extractinfo.Rmd" target="_blank">9907294</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-22
</td>
<td>
clean text
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SFonsecaCosta/Bgee/3118c12fb298dfed29663b63d4e757672d38fbc9/docs/extractinfo.html" target="_blank">3118c12</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SFonsecaCosta/Bgee/blob/4b9ee875926f5fcdce75eee565b4f2b458791149/analysis/extractinfo.Rmd" target="_blank">4b9ee87</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-20
</td>
<td>
cache=FALSE
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/SFonsecaCosta/Bgee/8d821e2b4d95ed6dcc78135555e80427a7c48dd9/docs/extractinfo.html" target="_blank">8d821e2</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SFonsecaCosta/Bgee/blob/9073f839d6190c28fa785eecc15e12e945fdc57b/analysis/extractinfo.Rmd" target="_blank">9073f83</a>
</td>
<td>
SFonsecaCosta
</td>
<td>
2020-04-20
</td>
<td>
add analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-the-packages" class="section level2">
<h2>Load the packages</h2>
<pre class="r"><code>library(BgeeDB)
library(here)</code></pre>
</div>
<div id="check-the-list-of-species" class="section level2">
<h2>Check the list of species</h2>
<p>Verify the list of available species in BgeeDB and the correspondent data types for each species.</p>
<pre class="r"><code>head(listBgeeSpecies())</code></pre>
<pre><code>
Querying Bgee to get release information...

Building URL to query species in Bgee release 14_1...

Submitting URL to Bgee webservice... (https://r.bgee.org/bgee14_1/?page=r_package&amp;action=get_all_species&amp;display_type=tsv&amp;source=BgeeDB_R_package&amp;source_version=2.12.1)

Query to Bgee webservice successful!</code></pre>
<pre><code>    ID          GENUS  SPECIES_NAME COMMON_NAME AFFYMETRIX   EST IN_SITU
1 6239 Caenorhabditis       elegans    nematode       TRUE FALSE    TRUE
2 7217     Drosophila     ananassae                  FALSE FALSE   FALSE
3 7227     Drosophila  melanogaster   fruit fly       TRUE  TRUE    TRUE
4 7230     Drosophila    mojavensis                  FALSE FALSE   FALSE
5 7237     Drosophila pseudoobscura                  FALSE FALSE   FALSE
6 7240     Drosophila      simulans                  FALSE FALSE   FALSE
  RNA_SEQ
1    TRUE
2    TRUE
3    TRUE
4    TRUE
5    TRUE
6    TRUE</code></pre>
</div>
<div id="check-list-of-bgee-releases" class="section level2">
<h2>Check list of Bgee releases</h2>
<p>You can verify the list of releases available in BgeeDB by using the <code>listBgeeRelease()</code> function.</p>
<pre class="r"><code>listBgeeRelease()</code></pre>
<pre><code>Downloading release information from Bgee...</code></pre>
<pre><code>  release release.date                        FTP.URL
1    14.1   2020-02-27 ftp://ftp.bgee.org/bgee_v14_1/
2    14.0   2018-02-21 ftp://ftp.bgee.org/bgee_v14_0/
3    13.2   2016-07-06 ftp://ftp.bgee.org/bgee_v13_2/
                   TopAnat.URL
1 https://r.bgee.org/bgee14_1/
2   https://r.bgee.org/bgee14/
3   https://r.bgee.org/bgee13/</code></pre>
<p>If you are interested in a particular release you can specify that by using the argument <code>release</code> and you are also able to order the table by a specific column, by speficying that in the argument <code>order</code>.</p>
<pre class="r"><code>head(listBgeeSpecies(release = &quot;13.2&quot;, order = 2))</code></pre>
<pre><code>
Querying Bgee to get release information...

Building URL to query species in Bgee release 13_2...

Submitting URL to Bgee webservice... (https://r.bgee.org/bgee13/?page=species&amp;display_type=tsv&amp;source=BgeeDB_R_package&amp;source_version=2.12.1)

Query to Bgee webservice successful!</code></pre>
<pre><code>      ID          GENUS SPECIES_NAME COMMON_NAME AFFYMETRIX   EST IN_SITU
17 28377         Anolis carolinensis      anolis      FALSE FALSE   FALSE
13  9913            Bos       taurus         cow      FALSE FALSE   FALSE
1   6239 Caenorhabditis      elegans   c.elegans       TRUE FALSE    TRUE
3   7955          Danio        rerio   zebrafish       TRUE  TRUE    TRUE
2   7227     Drosophila melanogaster    fruitfly       TRUE  TRUE    TRUE
5   9031         Gallus       gallus     chicken      FALSE FALSE   FALSE
   RNA_SEQ
17    TRUE
13    TRUE
1     TRUE
3    FALSE
2    FALSE
5     TRUE</code></pre>
</div>
<div id="retrieve-annotation" class="section level2">
<h2>Retrieve annotation</h2>
<p>In order to retrieve the annotations about the species and the data type of interest you have to create a Bgee object. In the <code>species</code> argument, you need to provide the name of the species or the NCBI taxonomic IDs.</p>
<p>For this part of the course we will use the <em>Drosophila melanogaster</em> data and the last release of Bgee (14.1).</p>
<pre class="r"><code>DrosMelRNASeq &lt;- Bgee$new(species = &quot;Drosophila_melanogaster&quot;, dataType = &quot;rna_seq&quot;)</code></pre>
<pre><code>
Querying Bgee to get release information...

Building URL to query species in Bgee release 14_1...

Submitting URL to Bgee webservice... (https://r.bgee.org/bgee14_1/?page=r_package&amp;action=get_all_species&amp;display_type=tsv&amp;source=BgeeDB_R_package&amp;source_version=2.12.1)

Query to Bgee webservice successful!

API key built: 342617eeb6e728567eeaf07855efc4fa274e6ad21dde329ecce7bd3668c4efa9f23c364808e45b7b772594a592dfe6855ff75027e7983a848ff43a25416ebe6f</code></pre>
<pre class="r"><code>annotation_DrosMelRNASeq &lt;- getAnnotation(DrosMelRNASeq)</code></pre>
<pre><code>
NOTE: annotation files for this species were found in the download directory /Users/sfonseca1/Bgee/Drosophila_melanogaster_Bgee_14_1. Data will not be redownloaded.</code></pre>
<p>The output folder from the download should be saved in your current directory, where you should have a folder called: SPECIES_Bgee_release (in this particular case: Drosophila_melanogaster_Bgee_14_1).</p>
</div>
<div id="target-experiments-from-annotation" class="section level2">
<h2>Target experiments from annotation</h2>
<p>From the information files (in this case: Drosophila_melanogaster_RNA-Seq_experiments.tsv or Drosophila_melanogaster_RNA-Seq_libraries.tsv) you are able to target experiments or libraries of interest before you start downloading the data.</p>
<p>Suggestion: Try to extract an experiment that have a minimum amount of 10 libraries and maximum of a 25 libraries in <em>Drosophila melanogaster</em> information file. During the filtering process allow the experiment to have samples in different conditions and samples in different stages.</p>
<pre class="r"><code>annExperiments &lt;- read.table(file = here(&quot;Drosophila_melanogaster_Bgee_14_1&quot;,&quot;Drosophila_melanogaster_RNA-Seq_experiments.tsv&quot;), header=T, sep=&quot;\t&quot;)

experimentsId &lt;- subset(annExperiments, Library.count &gt; 10 &amp; Library.count &lt; 25 &amp; Condition.count &gt; 5 &amp; Condition.count &lt; 15 &amp; Organ.stage.count &gt; 5)

## retrieve experiments 
unique(experimentsId)</code></pre>
<pre><code>  Experiment.ID
3     SRP002072
                                                                     Experiment.name
3 Developmental Profile of the Drosophila Transcriptome by Paired-End RNA-Sequencing
  Library.count Condition.count Organ.stage.count Organ.count Stage.count
3            12              10                 9           4           9
  Sex.count Strain.count Data.source                           Data.source.URL
3         2            0         SRA http://www.ncbi.nlm.nih.gov/sra/SRP002072
                                                                                                                                                Bgee.normalized.data.URL
3 ftp://ftp.bgee.org/bgee_v14_1/download/processed_expr_values/rna_seq/Drosophila_melanogaster/Drosophila_melanogaster_RNA-Seq_read_counts_TPM_FPKM_SRP002072.tsv.tar.gz
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Experiment.description
3 In order to gain a broad sampling of the Drosophila transcriptome, RNA-Seq experiments were performed at all stages of the Drosophila life cycle, and 12 independent cDNA libraries were generated, including embryonic, larval, pupal, and adult. Some libraries were staged as specific windows: 2-4 hour embryo, 14-16 hour embryo, 3rd instar larva, 3-day pupa, and 17-day adult. Additional libraries were derived from broadly staged mixed samples: embryo, larvae, and pupa. Three-day old male and female adults were sequenced separately for discovery of sex-specific variation. Finally, one library of mixed-age pupal RNA was sequenced in replicate as a validation of the technology. A total of 272 million paired-end reads of 64-75 base pairs in length were obtained, representing greater than 690x sequence coverage of the 30Mb Drosophila melanogaster transcriptome.</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.0 (2019-04-26)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] here_0.1             BgeeDB_2.12.1        tidyr_1.0.2         
 [4] topGO_2.38.1         SparseM_1.78         GO.db_3.10.0        
 [7] AnnotationDbi_1.48.0 IRanges_2.20.2       S4Vectors_0.24.4    
[10] Biobase_2.46.0       graph_1.64.0         BiocGenerics_0.32.0 
[13] workflowr_1.6.1     

loaded via a namespace (and not attached):
 [1] tidyselect_1.0.0   xfun_0.13          purrr_0.3.3        lattice_0.20-41   
 [5] vctrs_0.2.4        htmltools_0.4.0    yaml_2.2.1         blob_1.2.1        
 [9] rlang_0.4.5        later_1.0.0        pillar_1.4.3       glue_1.4.0        
[13] DBI_1.1.0          bit64_0.9-7        matrixStats_0.56.0 lifecycle_0.2.0   
[17] stringr_1.4.0      memoise_1.1.0      evaluate_0.14      knitr_1.28        
[21] httpuv_1.5.2       fansi_0.4.1        Rcpp_1.0.4.6       promises_1.1.0    
[25] backports_1.1.6    fs_1.4.1           bit_1.1-15.2       digest_0.6.25     
[29] stringi_1.4.6      dplyr_0.8.5        rprojroot_1.3-2    grid_3.6.0        
[33] cli_2.0.2          tools_3.6.0        bitops_1.0-6       magrittr_1.5      
[37] RCurl_1.98-1.1     RSQLite_2.2.0      tibble_3.0.0       crayon_1.3.4      
[41] whisker_0.4        pkgconfig_2.0.3    ellipsis_0.3.0     data.table_1.12.8 
[45] assertthat_0.2.1   rmarkdown_2.1      R6_2.4.1           git2r_0.26.1      
[49] compiler_3.6.0    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
