---
title: "Processing data"
author: 
- name: Sara Fonseca Costa
  affiliation:
  - &DEE Department of ecology and evolution, University of Lausanne & Swiss Institute of Bioinformatics, Switzerland
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
  html_document:
    toc: true
---

## Load the packages

```{r load-libs, message = FALSE, warning = FALSE, echo=TRUE}
library(BgeeDB)
library(ggplot2)
library(ggstatsplot)
library(varhandle)
library(here)
```


```{r include = FALSE}
knitr::opts_chunk$set(cache = FALSE, autodep = TRUE)
```

## Processing the data

When you collect data from Bgee database you can explore the data in multiple different ways to extract knowledge for your research.

For this part of the course (processing data) we will continue using the *Drosophila melanogaster* data.
We will plot an overview of the all RNA-Seq data available for this species and then we go more narrow until the comparison between 2 samples that just differ by sex.

### All RNA-Seq data

When we download/collect all data for a particular data type from Bgee you are working with processed data (e.g., TPMs for RNA-seq).

```{r all-rnaseq-data, message = FALSE, include=FALSE}
knitr::opts_chunk$set(echo = T,results = "hide")

DrosMelRNASeq <- Bgee$new(species = "Drosophila_melanogaster", dataType = "rna_seq")
dataBgee_DM <- getData(DrosMelRNASeq)
annotation_DrosMelRNASeq <- getAnnotation(DrosMelRNASeq)
```

From this processed data you can plot for each sample, referent to each experiment, the distribution of relative expression values. In the graphic bellow we plot the boxplot of log2(TPM) for all libraries that are present in Bgee for RNA-Seq data in *Drosophila melanogaster*.

```{r all-rnaseq-data-fig, fig.width=20,fig.height=8, warning = FALSE}
principalInfo <- c()
for (i in dataBgee_DM) {
  
  collect <- i[c(1,2,4,5,7,9, 10,12,14)]
  principalInfo <- rbind(principalInfo, collect)
}

ggplot(principalInfo, aes(x = Experiment.ID, y = log2(TPM))) +
        geom_boxplot(aes(color = Library.ID), show.legend = FALSE) + xlab("Experiments") + ylab("Log2(TPM)") +
        ggtitle("Libraries per experiments from RNA-Seq data")+
  scale_color_viridis_d() +
        theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

You can also group the data by corresponding anatomical entities. Note that here a violin_plot can represent multiple samples if the libraries in same experiment belongs to same the anatomical entity.

```{r all-rnaseq-data-anntEnt, fig.width=20,fig.height=8, warning = FALSE}
ggplot(principalInfo, aes(x = Experiment.ID, y = log2(TPM), fill = Anatomical.entity.ID)) +
        geom_violin() + xlab("Experiments") + ylab("Log2(TPM)") +
        ggtitle("Experiments per anatomical entity")+
  scale_color_viridis_d() +
        theme(axis.text.x = element_text(angle = 45, hjust = 1))+ guides(colour = guide_legend(override.aes = list(size=1)))
```

### Per experiment

Using as an example in the course, we will focus on the experiment retrieved in the section Extract Information, this means the experimentId SRP002072.

By plotting the data (just to visualize) you should be able to find the samples that belongs to the same anatomical entity and developmental stage for this experiment.

```{r process-one-experiment, warning=FALSE}
experiment_SRP002072 <- dplyr::filter(principalInfo, Experiment.ID == "SRP002072")

## remove useless columns in this case Experiment.ID = SRP002072 and Strain = NA
experiment_SRP002072[1] <- NULL
experiment_SRP002072[6] <- NULL

ggplot(experiment_SRP002072, aes(x = Library.ID, y = log2(TPM))) +
        geom_violin(aes(fill = Anatomical.entity.ID, col = Stage.ID)) + xlab("Libraries") + ylab("Log2(TPM)") +
        ggtitle("Experiment SRP002072")+
  scale_color_viridis_d() +
        theme(axis.text.x = element_text(angle = 45, hjust = 1))+ guides(colour = guide_legend(override.aes = list(size=1)))
```

From this experiment we can collect samples from the same anatomical entity and same developmental stage. Now we will search which samples from this experiment differ by sex.

```{r libraries-from-experiment-SRP002072}
sexInfo <- c("male", "female")
libInfo <- dplyr::filter(experiment_SRP002072, Sex %in% sexInfo)
cat(unique(libInfo$Library.ID))
```

From our filtering we know that the samples retrieved as unique have the same anatomical entity and same developmental stage but they differ in sex. 
Just verify how they correlate in overall (relative expression - TPM).


```{r libraries-correlation, warning=FALSE}
## extract just fundamental columns to make a reformated data.frame
libInfo <- libInfo[c(1,2,6)]

## reformat the data
reshapeTable <- reshape(libInfo, direction='wide',idvar='Library.ID', timevar='Gene.ID')
reshapeTable <- t(reshapeTable)[-c(1),]
finalTable <- data.frame(as.numeric(reshapeTable[,1]), as.numeric(reshapeTable[,2]))
colnames(finalTable) <- c("SRX019652", "SRX019653")
finalTable <- log(finalTable)
finalTable <- as.matrix(finalTable)

finalTable <- finalTable[!rowSums(!is.finite(finalTable)),]
finalTable <- as.data.frame(finalTable)

ggstatsplot::ggscatterstats(
  data = finalTable,
  x = SRX019652,
  y = SRX019653,
  xlab = "SRX019652 (F)",
  ylab = "SRX019653 (M)",
  title = "Correlation between samples from same experiment",
  messages = FALSE
)
```

Note that until this moment we are working and ploting the relative expression values of the genes across libraries, but this doens't mean that all of this genes were classified in Bgee database as present.

Using the same libraries that were used to make the correlation plot, we can explore the gene variability between this two samples by using just the genes that were classified by Bgee database as present. 

```{r gene-variability, warning=FALSE}
experiment_SRX019652 <- dplyr::filter(principalInfo, Experiment.ID == "SRP002072" & Library.ID == "SRX019652" & Detection.flag == "present")
experiment_SRX019652 <- experiment_SRX019652[c(3,8)]
colnames(experiment_SRX019652) <- c("Gene.ID", "TPM_SRX019652")

experiment_SRX019653 <- dplyr::filter(principalInfo, Experiment.ID == "SRP002072" & Library.ID == "SRX019653"& Detection.flag == "present")
experiment_SRX019653 <- experiment_SRX019653[c(3,8)]
colnames(experiment_SRX019653) <- c("Gene.ID", "TPM_SRX019653")

mergeLibs <- merge(experiment_SRX019652, experiment_SRX019653, by = "Gene.ID")
mergeLibs$meanSamples <- apply(mergeLibs[,2:3], 1, mean)
mergeLibs$sdSamples <- apply(mergeLibs[,2:3], 1, sd)

ggplot(mergeLibs, aes(x=log(meanSamples), y=log(sdSamples)))+
  geom_point(aes(color= ifelse(log(meanSamples)>8, "red", "blue")))+ labs(title="Gene variability",
                     x ="Log mean", y = "Log standard deviation")+ theme(legend.position = "none")+
  geom_vline(xintercept = 8, col="gray",linetype = "dashed")+geom_hline(yintercept = 5, col="gray",linetype = "dashed")
  
selectGenesDM <- dplyr::filter(mergeLibs, log(mergeLibs$meanSamples) > 8 & log(mergeLibs$sdSamples) > 5)
write.table(selectGenesDM$Gene.ID, file = here("genes_Drosophila_melanogaster.tsv"), sep="\t", col.names = FALSE, row.names = FALSE)
head(selectGenesDM)
```

From our analysis we can detected that 26 genes present high variability between the two samples.
This list of the genes can be used in analysis as TopAnat to verify if they enrich to specific target anatomical terms.

As summary, from this small analysis you were able to use the Bgee processed data to perform downstream analysis, such as correlation analysis or gene varialibility discovery across samples.
But with bgee data you can do even more such as clustering or differential expression (DE) analysis.
Normally for downstream analysis as DE the stand-alone tools expect counts as input, this information you can retrieve to all samples when you use the getData() function.

In order to get an overview of the entire experiment SRP002072 from where you collect the two sample above, perform a principal component analysis (PCA) for all libraries that belongs to this particular experiment using the raw counts. Note this can be done already with normalized values (TPM or FPKM).

```{r pca, warning=FALSE}
## collect raw counts for each library
principalInfo_raw <- c()
for (i in dataBgee_DM) {
  
  collect <- i[c(1,2,4,5,7,9, 10,11,14)]
  principalInfo_raw <- rbind(principalInfo_raw, collect)
}
## just present in bgee
experiment_SRP002072 <- dplyr::filter(principalInfo_raw, Experiment.ID == "SRP002072" & Detection.flag == "present")
libInfoData <- experiment_SRP002072[c(2,3,8,9)]
libInfoPresent <- libInfoData[c(1,2,3)]

reshapeTablePresentRaw <- reshape(libInfoPresent, direction='wide',idvar='Library.ID', timevar='Gene.ID')
allTablePresentRaw <- t(reshapeTablePresentRaw)

## clean data after transformation
colnames(allTablePresentRaw) <- NULL
colnames(allTablePresentRaw) <- allTablePresentRaw[1,]
allTablePresentRaw <- allTablePresentRaw[-1,]
allTablePresentRaw <- data.frame(allTablePresentRaw[,1:12])
allTablePresentRaw <- unfactor(allTablePresentRaw[,1:12])

## remove rows with NA
finalTableRaw <- na.omit(allTablePresentRaw)

## prepare final table to PCA
finalTableRaw <- t(finalTableRaw)

raw_pca <- prcomp(finalTableRaw)
percentage <- round(raw_pca$sdev / sum(raw_pca$sdev) * 100, 2)
percentage <- paste0(colnames(raw_pca$x), "(", as.character(percentage), "%)")

raw_pca <- as.data.frame(raw_pca$x)
raw_pca$Anatomical_Name <- c("embryo","embryo","embryo","embryo","larva","larva","larva", "pupa","pupa" ,"adult organism","adult organism","adult organism")

p<-ggplot(raw_pca,aes(x=PC1,y=PC2,color=Anatomical_Name, label = rownames(raw_pca) ))
p<-p+geom_point()+ xlab(percentage[1]) + ylab(percentage[2])+geom_text(size=2)+labs(title="Libraries from SRP002072 experiment")
p

```

## Cross annotation with processed data

After having explored the processed data from Bgee we can still cross information from the annotation with processed data.
From the output provided by the `getAnnotation()` function we can verify the amount of protein coding genes present, as well as other informations for each sample (e.g., genic region present per sample or percentage of intergenic present).

```{r plot-PC}
samplesId <- c(unique(libInfo$Library.ID))
infoLib <- dplyr::filter(annotation_DrosMelRNASeq$sample.annotation, Library.ID %in% samplesId)

ggplot(infoLib, aes(x = Library.ID, y = Protein.coding.genes.percent.present)) +
        geom_point() + ylim(0, 100) + xlab("Libraries") + ylab("% Protein coding") +
        ggtitle("Selected samples - SRX019652 (F) & SRX019653 (M)")
```


### Protein coding (All samples)

In the aforementioned graph, we just plotted the 2 samples from our narrow analysis. 

In order to get an overview of the % of protein coding genes for all samples from *Drosophila melanogaster* we should make a general plot.

```{r all-data-PC, fig.width=20,fig.height=10}
ggplot(annotation_DrosMelRNASeq$sample.annotation, aes(x = Experiment.ID, y = Protein.coding.genes.percent.present)) +
        geom_point(aes(color = Anatomical.entity.name ), show.legend = TRUE) + xlab("Experiments") + ylab("% Protein coding") +
        ggtitle("Percentage of protein coding per library")+
        theme(axis.text.x = element_text(angle = 45, hjust = 1))

```

